### PR TITLE
Better support for default interface and explicitly implemented properties

### LINKED ISSUE #2754: BeEquivalentTo doesn't always choose normal properties over explicitly implemented properties
### Description

I use my own implementation of Exception which implements a interface with a Message-Property. This property is of another type.

### Reproduction Steps

```csharp
[Fact]
public void Test()
{
    var instance = new MyClass()
    {
        MyError = 42,
    };
    var other = new MyClass()
    {
        MyError = 42,
    };

    instance.Should().BeEquivalentTo(other);
}

public class MyClass : Exception, IMyInterface
{
    public int MyError { get; set; }
    int IMyInterface.Message => MyError;
}

public interface IMyInterface
{
    int Message { get; }
}
```

### Expected behavior

The test passes.

### Actual behavior

The test failes with the message:

```
Message: 
Expected property instance.MyNamespace.IMyInterface.Message to be System.String, but found System.Int32.

With configuration:
- Use declared types and members
- Compare enums by value
- Compare tuples by their properties
- Compare anonymous types by their properties
- Compare records by their members
- Include non-browsable members
- Match member by name (or throw)
- Be strict about the order of items in byte arrays
- Without automatic conversion.


  Stack Trace: 
XUnit2TestFramework.Throw(String message)
TestFrameworkProvider.Throw(String message)
CollectingAssertionStrategy.ThrowIfAny(IDictionary`2 context)
AssertionScope.Dispose()
EquivalencyValidator.AssertEquality(Comparands comparands, EquivalencyValidationContext context)
ObjectAssertions`2.BeEquivalentTo[TExpectation](TExpectation expectation, Func`2 config, String because, Object[] becauseArgs)
ObjectAssertions`2.BeEquivalentTo[TExpectation](TExpectation expectation, String because, Object[] becauseArgs)
MyTestClass.Test() line 426
RuntimeMethodHandle.InvokeMethod(Object target, Void** arguments, Signature sig, Boolean isConstructor)
MethodBaseInvoker.InvokeWithNoArgs(Object obj, BindingFlags invokeAttr)
```

### Regression?

In 6.12.0 it worked by with 6.12.1 it is failing.

### Known Workarounds

_No response_

### Configuration

net8.0
FluentAssertions 6.12.1

### Other information

_No response_

### Are you willing to help with a pull-request?

No

### PR DESCRIPTION
**Pull request by dennisdoomen**
Reimplemented the Reflection process needed to find all types of properties and fields including default interface properties.

Fixes #2754 and #2759

## IMPORTANT 

* [ ] If the PR touches the public API, the changes have been approved in a separate issue with the "api-approved" label.
* [ ] The code complies with the [Coding Guidelines for C#](https://www.csharpcodingguidelines.com/).
* [ ] The changes are covered by unit tests which follow the Arrange-Act-Assert syntax and the naming conventions such as is used [in these tests](../tree/develop/Tests/FluentAssertions.Equivalency.Specs/MemberMatchingSpecs.cs#L51-L430).
* [ ] If the PR adds a feature or fixes a bug, please update [the release notes](../tree/develop/docs/_pages/releases.md) with a functional description that explains what the change means to consumers of this library, which are published on the [website](https://fluentassertions.com/releases).
* [ ] If the PR changes the public API the changes needs to be included by running [AcceptApiChanges.ps1](../tree/develop/AcceptApiChanges.ps1) or [AcceptApiChanges.sh](../tree/develop/AcceptApiChanges.sh).
* [ ] If the PR affects [the documentation](../tree/develop/docs/_pages), please include your changes in this pull request so the documentation will appear on the [website](https://www.fluentassertions.com/introduction).
    * [ ] Please also run `./build.sh --target spellcheck` or `.\build.ps1 --target spellcheck` before pushing and check the good outcome

### CODE REVIEW

File: Src/FluentAssertions/Common/TypeMemberReflector.cs [hunk 1]
```diff
@@ -11,159 +11,137 @@ namespace FluentAssertions.Common;
...
+        return prop.Name.Contains('.', StringComparison.OrdinalIgnoreCase);
+    }
 
-        while (typeToReflect != null)
+    private void LoadFields(Type typeToReflect, MemberVisibility visibility)
+    {
+        while (typeToReflect is not null && typeToReflect != typeof(object))
         {
-            foreach (var memberInfo in getMembers(typeToReflect))
+            FieldInfo[] files = typeToReflect.GetFields(
```
Comment (IT-VBFK) [inline Src/FluentAssertions/Common/TypeMemberReflector.cs:121]: `files`? Maybe a typo?

**Review by IT-VBFK — COMMENTED** [review]
🤔 Missing release notes?

**Comment by IT-VBFK** [comment]
What's wrong with this to be closed?

**Comment by dennisdoomen** [comment]
I'm going to fix it in a different way.

**Pull request closed without merging**